src/jsonrpc/miner_jsonrpc_txn.erl

```diff
@@ -0,0 +1,86 @@
+-module(miner_jsonrpc_txn).
+
+-include("miner_jsonrpc.hrl").
+-behavior(miner_jsonrpc_handler).
+
+%% jsonrpc_handler
+-export([handle_rpc/2]).
+
+%%
+%% jsonrpc_handler
+%%
+
+handle_rpc(<<"txn_queue">>, []) ->
+    case (catch blockchain_txn_mgr:txn_list()) of
+        {'EXIT', _} -> #{ error => <<"timeout">> };
+        [] -> [];
+        Txns ->
+            maps:fold(fun(T, D, Acc) ->
+                              Type = blockchain_txn:type(T),
+                              Hash = blockchain_txn:hash(T),
+                              Accepts = proplists:get_value(acceptions, D, []),
+                              Rejects = proplists:get_value(rejections, D, []),
+                              AcceptHeight = proplists:get_value(recv_block_height, D, undefined),
+                              [ #{ type => Type,
+                                   hash => ?BIN_TO_B64(Hash),
+                                   accepts => length(Accepts),
+                                   rejections => length(Rejects),
+                                   accepted_height => AcceptHeight } | Acc ]
+                      end, [], Txns)
+    end;
+handle_rpc(<<"txn_add_gateway">>, #{ owner := OwnerB58 } = Params) ->
+    try
+        Payer = case maps:get(payer, Params, undefined) of
+                    undefined -> undefined;
+                    PayerB58 -> ?B58_TO_BIN(PayerB58)
+                end,
+        {ok, Bin} = blockchain:add_gateway_txn(?B58_TO_BIN(OwnerB58), Payer),
+        B64 = base64:encode(Bin),
+        #{ <<"result">> => B64 }
+    catch
+        T:E:St ->
+            lager:error("Couldn't do add gateway via JSONRPC because: ~p ~p: ~p",
+                        [T, E, St]),
+            Error = io_lib:format("~p", [E]),
+            #{ <<"error">> => Error }
+    end;
+handle_rpc(<<"txn_assert_location">>, #{ owner := OwnerB58,
+                                         location := Loc } = Params) ->
```

Comment:
use `lat` and `lon` arguments instead of string splitting

Reply:
Yeah good ideas